sdk/resource_locator.go

```diff
@@ -20,18 +20,30 @@ import (
 // ResourceLocator - structure to contain a protocol + body comprising an URL
 type ResourceLocator struct {
 	protocol urlProtocol // See urlProtocol values below
-	body     string      // Body of url
+	// body URL without protocol scheme
+	body string
+	// identifier unique to this URL
+	identifier string
 }
 
 // urlProtocol - shorthand for protocol prefix on fully qualified url
+// also specifies the optional resource identifier - current usage is a key identifier
 type urlProtocol uint8
 
 const (
-	kMaxBodyLen      int         = 255
+	kMaxBodyLen       int = 255
+	kMaxIdentifierLen int = 32
+	// kPrefixHTTPS identifier field is size of 0 bytes (not present)
 	kPrefixHTTPS     string      = "https://"
 	kPrefixHTTP      string      = "http://"
 	urlProtocolHTTP  urlProtocol = 0
 	urlProtocolHTTPS urlProtocol = 1
+	// urlProtocolHTTPSID2B identifier field is size of 2 bytes
+	urlProtocolHTTPSID2B urlProtocol = 33
```

Comment:
See my note https://github.com/opentdf/spec/pull/40#discussion_r1695814703 about separating this into two fields that are combined with a mask into a single byte

Reply:
updated